server/image_pull.go

```diff
@@ -20,10 +21,43 @@ import (
 
 	"github.com/cri-o/cri-o/internal/log"
 	"github.com/cri-o/cri-o/internal/storage"
+	"github.com/cri-o/cri-o/pkg/config"
 	"github.com/cri-o/cri-o/server/metrics"
 	"github.com/cri-o/cri-o/utils"
 )
 
+// GetRuntimeHandlerForPod returns the runtime handler for the given pod config.
+func (s *Server) GetRuntimeHandlerForPod(ctx context.Context, cfg *types.PodSandboxConfig) (string, *config.RuntimeHandler) {
```

Comment:
nit: can this be defined below where it's called.
does this need to be exported?